### PR TITLE
Few changes to the instructions how to compile a new kernel

### PR DESCRIPTION
**Pull request by halcon74**
Hi Ramon,

I am learning to make PRs... :)

I tried to make this PR especially for cherry-picking and, I need to say, it's not that easy.

I think in this case the best approach can be:

1) To write the complete new version of the file separately
2) Copy/Paste the lines from up to bottom, trying to split changes to the proper blocks, committing each block

And even then there will arise conflicts. For example, I rename a section and move this section up. OK, I split it to two steps... But if you like to merge the second step, not first, there will be a conflict :)

EDIT: IMHO, it contradicts the essence of a VCS; it's better to make multiple PRs.

### CODE REVIEW

**Comment by halcon74** [comment]
How did you submit the reviews?

File: usr/src/README [hunk 1]
```diff
@@ -41,16 +51,16 @@ compile new kernel
 
         6.3. check, if the microcode was updated
             dmesg | grep "microcode"
+	    
+    7. re-compile virtualbox and other modules for the new kernel:
+        emerge @module-rebuild
 
-    7. update the grub configuration:
+    8. update the grub configuration:
```
Comment (halcon74) [inline usr/src/README:58]: I have Grub2 on some my machines, and Grub legacy on others. Do you have Grub2 on all machines?

**Comment by halcon74** [comment]
Experimenting: I used the same blue button on your commit, adding this README, and it resulted in a similar review.

**Comment by halcon74** [comment]
A note: the review that I submitted for the lines about Grub, is shown not only in that commit where I submitted the review, but in the last one too (because one of the lines about Grub is shown there too) :) GitHub whistles :)

**Comment by keks24** [comment]
> And even then there will arise conflicts.

These might happen. I am going to fix them, since I am the owner of the repository.

> EDIT: IMHO, it contradicts the essence of a VCS; it's better to make multiple PRs.

I do not think so. Every commit is documented and I can choose to take all changes, pick some of them or reject everything.

Other pull requests would make sense, if it refers to another issue or topic.

Apropos `issues`: It is always good to report an `issue` and combine it with a `pull request`. Good examples I made so far:

* Gitea
    * [Issue](https://github.com/go-gitea/gitea/issues/8803)
    * [Pull request](https://github.com/go-gitea/gitea/pull/8804)
* Ansible (without pull request, since I do not know how to program in Python properly)
    * [Issue](https://github.com/ansible/ansible/issues/28772)
    * [Repository to reproduce the issue](https://github.com/keks24/ansible-issue-28772)

Edit - 20200921:
Oh no. By adding the links of the issues here, GitHub automatically referenced them. Shit happens. :D

By the way: Since this is a mirror repository from my internal Gitea instance, I am going to do some experiments here and then adapt the changes locally, so your pull request might disappear here.

File: usr/src/README (hunk 1 above)
Comment (keks24) [reply usr/src/README:58]: Marking the next line would be more explainable. :) Yes, I use `GRUB2` on all of my systems. It is to mention, that this repository is for my own usage only, so I did not cover any other possibility. I think, every big project starts with something like this and grows, when others have a use of it and then run into issues.

**Comment by keks24** [comment]
Concerning "how to checkout a pull request": On the upper right corner, there is a button called `Open with`. You can then check out the pull request via `gh pr checkout <pull_request_number>`. I totally forgot, that GitHub recently implemented a feature to manage GitHub repositories via CLI.

File: usr/src/README (hunk 1 above)
Comment (halcon74) [reply usr/src/README:58]: > Marking the next line would be more explainable. :)

Agreed! :)

> Yes, I use `GRUB2` on all of my systems. It is to mention, that this repository is for my own usage only, so I did not cover any other possibility.

Yes, sure. No reason to complicate a manual by unsupported options.

> I think, every big project starts with something like this and grows, when others have a use of it and then run into issues.

You're right. It's logically consistent.

**Comment by halcon74** [comment]
> Apropos `issues`: It is always good to report an `issue` and combine it with a `pull request`. Good examples I made so far:

Thanks. I'll keep it in mind.

> Edit - 20200921:
> Oh no. By adding the links of the issues here, GitHub automatically referenced them. Shit happens. :D

I'll keep it in mind too :D

> By the way: Since this is a mirror repository from my internal Gitea instance, I am going to do some experiments here and then adapt the changes locally, so your pull request might disappear here.

OK.

**Comment by halcon74** [comment]
> Concerning "how to checkout a pull request": On the upper right corner, there is a button called `Open with`. You can then check out the pull request via `gh pr checkout <pull_request_number>`. I totally forgot, that GitHub recently implemented a feature to manage GitHub repositories via CLI.

Good to know, about 'gh'!

**Comment by keks24** [comment]
Here is what I did with your changes:

~0. Simulating, that I work via `GitHub`:~
```bash
# this generates a random directory and automatically navigates to it "cd $(mktemp -d)"; see 1.
$ cdt
$ git clone "git@github.com:keks24/dotfiles.git"
$ cd "dotfiles/"
```

0. Since I am working with my local `Gitea` instance, I add a separate origin `github`:
```bash
$ git checkout master
$ git remote add github "git@github.com:keks24/dotfiles.git"
$ git remote -v
github  git@github.com:keks24/dotfiles.git (fetch)
github  git@github.com:keks24/dotfiles.git (push)
origin  gitea@gitea.local:keks24/dotfiles.git (fetch)
origin  gitea@gitea.local:keks24/dotfiles.git (push)
```

1. Check out this pull request and automatically create a new branch, called `enhancements_compiling_kernel`<sup>2</sup>:
```bash
$ git fetch github pull/1/head:enhancements_compiling_kernel
```

This avoids to apply the changes to the current branch I am working on, which is currently `master`. 

Apropos `branch names`: GitHub is about to change the branch name `master` to `main`. This `anti-racism` thing is going around like fire. At many points this is so ridiculous and it will only cause more issues...

2. Check out the new branch:
```bash
$ git branch
enhancements_compiling_kernel
master
$ git checkout enhancements_compiling_kernel
```

3. Take a look at the log:
```bash
$ git log
# the best cli tool, i found so far, when working with git repositories; see 3 and 4.
$ tig
```

I see here, that you state what you have changed. I see this in the diff, so this is actually redundant. It would be better to tell why you have changed it. [For example:](https://github.com/keks24/dotfiles/pull/1/commits/dbdf88b8d3fd68319f9b6afd77b880f67d26187b)

Rename

`Rename Section 1.2 to 1.1`

to

`correct typo`

Associating the commit message `correct typo` with your change makes the explanation complete. A commit message should always answer the question `why` and not `what`.

4. Edit and pick<sup>5</sup> some of the commits and apply them:
```bash
$ git log --pretty="oneline"
$ git checkout master
# cherry picking from the branch "enhancements_compiling_kernel"
$ git cherry-pick dbdf88b8d3fd68319f9b6afd77b880f67d26187b
[...]
# when there are some conflicts, edit the file with an editor and then
$ git add usr/src/README
$ git cherry-pick --continue
```

Ah, I see. Cherry picking is also `merging` at the same time, where you also might have to solve `merge conflicts`. Now I understand your issues with my pull request. This is actually for advanced users. And this<sup>7</sup> is how to abort a `git cherry-pick`.

I cherry picked some of your commits and applied my changes to them.

I also stumbled across this page<sup>6</sup>, which nicely adapts the output of `git log`.

5. Push all changes:
```bash
$ git push
```

6. Remove the origin `github`:
```bash
$ git remote remove github
$ git remote -v
origin  gitea@gitea.local:keks24/dotfiles.git (fetch)
origin  gitea@gitea.local:keks24/dotfiles.git (push)
```

<sup>1</sup> https://github.com/grml/grml-etc-core/blob/v0.17.4/etc/zsh/zshrc#L3419
<sup>2</sup> https://stackoverflow.com/a/30584951
<sup>3</sup> https://github.com/jonas/tig
<sup>4</sup> 20200921-121934_screenshot
<sup>5</sup> https://mattstauffer.com/blog/how-to-merge-only-specific-commits-from-a-pull-request/
<sup>6</sup> https://ma.ttias.be/pretty-git-log-in-one-line/
<sup>7</sup> https://stackoverflow.com/a/16826728

**Comment by halcon74** [comment]
Thanks for sharing your experience! I'll study it.

Mercurial (which I'm used to) branches are designed in somewhat another way. There are multiple heads, which can diverge-converge. (One commit can't be included in different branches)

**Comment by halcon74** [comment]
Hi Ramon, I noticed also that there is not mentioned `cd linux` before `make` (I see no reason to create another PR for it)

**Comment by keks24** [comment]
> Hi Ramon, I noticed also that there is not mentioned `cd linux` before `make` (I see no reason to create another PR for it)

Oh yeah, I forgot to add that!

You can still work with your `fork` and adapt it. The commit message will be shown here.

**Comment by keks24** [comment]
Done; see https://github.com/keks24/dotfiles/pull/1#issuecomment-696036428.

Thanks for the effort!

**Comment by halcon74** [comment]
Thanks for the information about `cherry-pick` (`abort`, `continue`) and other details. 

About editing the file with an editor: my editor for git wasn't configured properly and wasn't opening (now it is).

>You can still work with your fork and adapt it. The commit message will be shown here.

Shown where? I'm going to check it...

**Comment by keks24** [comment]
> About editing the file with an editor: my editor for git wasn't configured properly and wasn't opening (now it is).

I meant this:

https://github.com/halcon74/dotfiles/pull/1#issuecomment-692967997

You can edit that with any editor to solve `merge conflicts`. :)

> Shown where? I'm going to check it...

I meant, if you do further editing on your `fork` and push it to your `fork`, all commit messages are shown in this pull request like this:

20200924-001456_screenshot

**Comment by halcon74** [comment]
I added a new commit and pushed it to origin, but I don't see it here...

https://github.com/halcon74/dotfiles-1/commit/23010b979c8df62465369821877585f698de76ae

Maybe, it's because the PR is closed?

**Comment by keks24** [comment]
Yes, that could be it. You could test it with your other pull request about `g-cpan`.

**Comment by halcon74** [comment]
Possible. If/when I have something to add. Does it work only for master branch?

**Comment by keks24** [comment]
I never tested it, but I think, it depends on which branch of yours you want to merge. In your case, it is `master`.

20200924-114004_screenshot

**Pull request closed without merging**